### PR TITLE
fix(amf): encoding and decoding functions for MaxSuppPktFilterIe

### PR DESCRIPTION
**Pull request by VinayKumar1591**
fix(amf): encoding and decoding functions for MaxSuppPktFilterIe
## Summary

1. Added encoding/decoding to support MaxNumOfSupportedFiltersIE
2. Added UT to test decoding functions

## Test Plan

1. Tested on NGAP Tester

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMaxNumOfSupportedPacketFilters.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+#include <iostream>
+#include <sstream>
+#include <cstdint>
+#include <cstring>
+#include "M5GCommonDefs.h"
+#include "M5GMaxNumOfSupportedPacketFilters.h"
+
+namespace magma5g {
+M5GMaxNumOfSupportedPacketFilters::M5GMaxNumOfSupportedPacketFilters(){};
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMaxNumOfSupportedPacketFilters.cpp]: legit github actions comments

Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMaxNumOfSupportedPacketFilters.cpp]: Addressed

**Review by ardzoht — APPROVED** [review]
lgtm

**Comment by ulaskozat** [comment]
clang format check is failing. please run the formatter in dev vm and push new commit.

**Comment by VinayKumar1591** [comment]
> clang format check is failing. please run the formatter in dev vm and push new commit.

Addressed clang format check issues

**Pull request merged**